contrib/opentimelineio_contrib/adapters/otio_svg.py

```diff
@@ -0,0 +1,481 @@
+                   r' style="stroke:{};stroke-width:{};' \
+                   r'fill:{};opacity:{};">{}</text>'.format(text_size, self.font_family,
+                                                            location_svg.x,
+                                                            location_svg.y,
+                                                            self.svg_color(
+                                                                color),
+                                                            stroke_width / 4.0,
+                                                            self.svg_color(
+                                                                color), color.a, text)
+        self.lines.append(text_str)
+
+    def get_text_layout_size(self, text='', text_size=10.0):
+        width = 0.0
+        for character in text:
+            ascii_val = int(ord(character))
+            width += self.LCARS_CHAR_SIZE_ARRAY[ascii_val] if ascii_val < 128 else 0
+        scale_factor = text_size / 100.0
+        return width * scale_factor + 25
+
+    def save_image(self, file_path):
+        header = r'<svg height="{}" width="{}" version="4.0"' \
+                 r' xmlns="http://www.w3.org/2000/svg"' \
+                 r' xmlns:xlink= "http://www.w3.org/1999/xlink">'.format(self.height,
+                                                                         self.width)
+        background = r'<rect width="100%" height="100%" fill="white"/>'
+        font = r'<defs><style>@import url("https://fonts.googleapis.com/css?' \
+               r'family=\#(fontFamily)");</style></defs>'
+        separator = '\n'
+        image = header + '\n' + background + '\n' + font + '\n' + separator.join(
+            self.lines) + '\n</svg>'
+        svg_file = open(file_path, 'w')
+        svg_file.write(image)
+        svg_file.close()
+
+
+class ClipData:
+    src_start = 0.0
+    src_end = 0.0
+    avlbl_start = 0.0
+    avlbl_end = 0.0
+    avlbl_duration = 0.0
+    trim_start = 0.0
+    trim_duration = 0.0
+    target_url = ''
+    clip_id = 0
+
+    def __init__(self, src_start=0.0, src_end=0.0, avlbl_start=0.0,
+                 avlbl_end=0.0, avlbl_duration=0.0,
+                 trim_start=0.0, trim_duration=0.0, target_url='', clip_id=0):
+        self.src_start = src_start
+        self.src_end = src_end
+        self.avlbl_start = avlbl_start
+        self.avlbl_end = avlbl_end
+        self.avlbl_duration = avlbl_duration
+        self.trim_start = trim_start
+        self.trim_duration = trim_duration
+        self.target_url = target_url
+        self.clip_id = clip_id
+
+
+def convert_otio_to_svg(timeline, filepath):
+    renderer = SVGRenderer(2406.0, 1054.0)
+    image_margin = 10.0
+    font_size = 15
+    total_duration = 0
+    min_time = 0.0
+    max_time = 0.0
+    all_clips_data = []
+    clip_count = -1
+    for curr_clip in timeline.tracks[0]:
+        if isinstance(curr_clip, otio.schema.Clip):
```

Comment:
> One more question - instead of just iterating over track[0], can this algorithm be extended to work across all the tracks in a stack?

Yes, this is something that I wanted to discuss before moving forward. How do we want to handle multiple tracks? From what I understand, are tracks similar to the layers that we have in Adobe Premiere Pro? One way could be to increase the width of the image and have multiple tracks side by side. But this would probably make readability a bit difficult if we have very long `available` clips.
We could also place new tracks vertically below. Do you have any other way of doing this?

Reply:
Currently the multiple track document is here: 

![](https://opentimelineio.readthedocs.io/en/latest/_images/multiple_tracks.png)

@jminor might have another idea?

Reply:
Thanks! This is very helpful.